hw03/src/main/java/ru/otus/runner/TestRunner.java

```diff
@@ -0,0 +1,71 @@
+package ru.otus.runner;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import ru.otus.annotation.After;
+import ru.otus.annotation.Before;
+import ru.otus.annotation.Test;
+import ru.otus.test.TestClass;
+
+public class TestRunner {
+    private static final String EXCEPTION_ERR_MSG = "%s %s: %s";
+
+    private final String testClassName;
+
+    public TestRunner(String testClassName) {
+        this.testClassName = testClassName;
+    }
+
+    public void run() throws ClassNotFoundException {
+        Class<?> clazz = Class.forName(testClassName);
+        Method[] methods = clazz.getDeclaredMethods();
+        List<Method> beforeMethods = filterMethodsByAnnotation(methods, Before.class.getName());
+        List<Method> testMethods = filterMethodsByAnnotation(methods, Test.class.getName());
+        List<Method> afterMethods = filterMethodsByAnnotation(methods, After.class.getName());
```

Comment:
Лучше здесь сделать так:
```suggestion
        List<Method> beforeMethods = filterMethodsByAnnotation(methods, Before.class);
        List<Method> testMethods = filterMethodsByAnnotation(methods, Test.class);
        List<Method> afterMethods = filterMethodsByAnnotation(methods, After.class);
```
Передавать аннотацию не как строку, а как класс.
Соответствующим образом изменить `filterMethodsByAnnotation()`

Reply:
поправил